### PR TITLE
Add com.github.dubstar.design manifest

### PR DESCRIPTION
**Pull request by dubstar-04**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: wood.dan.design.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+    "app-id" : "wood.dan.design",
```
Comment (hfiguiere) [inline wood.dan.design.json:2]: `dan.wood` doesn't exist. I don't even think `wood` is a valid TLD.

File: wood.dan.design.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+      "name": "blueprint",
```
Comment (hfiguiere) [inline wood.dan.design.json:27]: this needs a cleaniup rule to remove it completely as it is a build tool.

File: wood.dan.design.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      ]
+    	},
+        {
+            "name" : "design",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/dubstar-04/Design.git"
```
Comment (hfiguiere) [inline wood.dan.design.json:45]: this needs a spec revision (tag or commit)

File: wood.dan.design.json (hunk 2 above)
Comment (dubstar-04) [reply wood.dan.design.json:27]: Do you know of any examples for this?

Comment (dubstar-04) [reply wood.dan.design.json:27]: just add the path to the cleanup section?

**Comment by dubstar-04** [comment]
If I have correctly fixed the issues I would like this added to flathub-beta

File: com.github.dubstar.design.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{
+    "app-id" : "com.github.dubstar.design",
```
Comment (hfiguiere) [inline com.github.dubstar.design.json:2]: to should be `io.github`

https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: com.github.dubstar.design.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    	},
+        {
+            "name" : "design",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "path" : "https://github.com/dubstar-04/Design.git",
+                    "tag": "/v/Design/43-alpha1",
```
Comment (hfiguiere) [inline com.github.dubstar.design.json:47]: this not a valid tag.

`git tag -l` will tell you

File: com.github.dubstar.design.json (hunk 4 above)
Comment (hfiguiere) [reply com.github.dubstar.design.json:2]: Also the org / user name being what it is, the acceptable app-id should be `io.github.dubstar_04.design`

**Comment by dubstar-04** [comment]
> The metainfo validation fails on the screenshot urls. image
> 
> The links are valid. Is there an issue with github urls?

This is now resolved

**Comment by hfiguiere** [comment]
these URL point to the page, not the raw file.

**Comment by hfiguiere** [comment]
bot, build io.github.dubstar_04.design

File: io.github.dubstar_04.design.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      ]
+    	},
+        {
+            "name" : "design",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "path" : "https://github.com/dubstar-04/Design.git",
```
Comment (hfiguiere) [inline io.github.dubstar_04.design.json:46]:
```suggestion
                    "url" : "https://github.com/dubstar-04/Design.git",
```

**Comment by dubstar-04** [comment]
Is there anything I need to do to progress this?

**Comment by barthalion** [comment]
bot, build io.github.dubstar_04.design

**Comment by barthalion** [comment]
This fails on `Check for 128x128 icon` now. Typically, you need to cross-check 3 things: whether there's an icon in somewhere `/app/share/`, the content of metainfo file and the desktop file. The metainfo reads the `Icon` field in the desktop file, and then looks for the file itself. It has to match the app ID.

**Comment by hfiguiere** [comment]
The launchable is wrong 
https://github.com/dubstar-04/Design/blob/43-alpha1/data/io.github.dubstar_04.design.metainfo.xml

I'm pretty sure it is not set properly at build time since this is not a `.in` file.

**Comment by dubstar-04** [comment]
> The launchable is wrong https://github.com/dubstar-04/Design/blob/43-alpha1/data/io.github.dubstar_04.design.metainfo.xml
> 
> I'm pretty sure it is not set properly at build time since this is not a `.in` file.

The launchable has to match the desktop filename? with '.in' included

**Comment by hfiguiere** [comment]
I'm pretty it's unchanged at installation

image

it it's wrong.

**Comment by hfiguiere** [comment]
actually it duplicates the one at the top. which is probably why it fails.

**Comment by hfiguiere** [comment]
on a side note moving tags is a bad idea. and a tag that that the same name as a branch, also a bad idea.

**Comment by dubstar-04** [comment]
Are we able to try building again?

**Comment by dubstar-04** [comment]
bot, build io.github.dubstar_04.design

**Comment by hfiguiere** [comment]
> @hfiguiere hopefully thats fixed now. Thank you for all your help.

you removed the wrong one. So it still fails.

**Comment by dubstar-04** [comment]
> > @hfiguiere hopefully thats fixed now. Thank you for all your help.
> 
> you removed the wrong one. So it still fails.

I don't understand. I renamed the desktop file and removed the duplicate launchable tag. The error is regarding a missing icon?

**Comment by hfiguiere** [comment]
You removed this:

image


This one is correct. The other one is not.

**Comment by dubstar-04** [comment]
bot, build io.github.dubstar_04.design

**Comment by dubstar-04** [comment]
@hfiguiere thank you for all your help. Really appreciated.

**Comment by dubstar-04** [comment]
Installs and works as expected! 

image

**Pull request closed without merging**